src/langchain_google_spanner/loader.py

```diff
@@ -0,0 +1,422 @@
+        """
+        instance = self.client.instance(self.instance_id)
+        db = instance.database(self.database_id)
+        timestamp = (
+            self.staleness.replace(tzinfo=datetime.timezone.utc)
+            if type(self.staleness) is datetime.datetime
+            else None
+        )
+        duration = (
+            datetime.timedelta(seconds=self.staleness)
+            if type(self.staleness) is float
+            else None
+        )
+        snapshot = db.batch_snapshot(exact_staleness=duration, read_timestamp=timestamp)
+        partitions = snapshot.generate_query_batches(
+            sql=self.query, data_boost_enabled=self.databoost
+        )
+        for partition in partitions:
+            r = snapshot.process_query_batch(partition)
+            results = r.to_dict_list()
+            if len(results) == 0:
+                break
+            column_names = [f.name for f in r.fields]
+            content_columns = self.content_columns or [column_names[0]]
+            metadata_columns = self.metadata_columns or [
+                col for col in column_names if col not in content_columns
+            ]
+            for row in results:
+                yield _load_row_to_doc(
+                    self.format,
+                    content_columns,
+                    metadata_columns,
+                    self.metadata_json_column,
+                    row,
+                )
+
+
+class SpannerDocumentSaver:
+    """Save docs to Google Cloud Spanner."""
+
+    def __init__(
+        self,
+        instance_id: str,
+        database_id: str,
+        table_name: str,
+        content_column: str = CONTENT_COL_NAME,
+        metadata_columns: List[str] = [],
+        metadata_json_column: str = METADATA_COL_NAME,
+        client: Optional[Client] = None,
+    ):
+        """Initialize Spanner document saver.
+
+        Args:
+            instance_id: The Spanner instance to load data to.
+            database_id: The Spanner database to load data to.
+            table_name: The table name to load data to.
+            content_column: The name of the content column. Defaulted to the first column.
+            metadata_columns: This is for user to opt-in a selection of columns to use. Defaulted to use
+                              all columns.
+            metadata_json_column: The name of the special JSON column. Defaulted to use "langchain_metadata".
+            client: The connection object to use. This can be used to customized project id and credentials.
+        """
+        self.instance_id = instance_id
+        self.database_id = database_id
+        self.table_name = table_name
+        self.content_column = content_column
+        self.metadata_columns = metadata_columns
+        self.metadata_json_column = metadata_json_column
+        self.client = client_with_user_agent(client, USER_AGENT_LOADER)
+        instance = self.client.instance(instance_id)
+        if not instance.exists():
```

Comment:
nit: can you please leave a one link blank before and after the conditional block for better readability?